core/services/ocr2/delegate.go

```diff
@@ -247,66 +248,62 @@ func (d *Delegate) OnDeleteJob(jb job.Job, q pg.Queryer) error {
+	rid, err := spec.RelayIdentifier()
 	if err != nil {
-		d.lggr.Errorw("OCR2 chainID not evm compatible", "err", err)
+		d.lggr.Errorw("DeleteJob: "+ErrJobSpecNoRelayer.Error(), "err", err)
 		return nil
 	}
-	chain, err := d.legacyChains.Get(chainID.String())
-	if err != nil {
-		d.lggr.Error(err)
-		return nil
-	}
-	lp := chain.LogPoller()
+	// only do cleanup for EVM
+	if rid.Network == relay.EVM {
 
-	var filters []string
-	switch spec.PluginType {
-	case job.OCR2VRF:
-		filters, err = ocr2coordinator.FilterNamesFromSpec(spec)
+		chain, err := d.legacyChains.Get(rid.ChainID.String())
 		if err != nil {
-			d.lggr.Errorw("failed to derive ocr2vrf filter names from spec", "err", err, "spec", spec)
+			d.lggr.Error("DeleteJob: failed to chain get chain %s", "err", rid.ChainID, err)
+			return nil
 		}
-	case job.OCR2Keeper:
-		filters, err = ocr2keeper.FilterNamesFromSpec20(spec)
-		if err != nil {
-			d.lggr.Errorw("failed to derive ocr2keeper filter names from spec", "err", err, "spec", spec)
+		lp := chain.LogPoller()
+
+		var filters []string
+		switch spec.PluginType {
+		case job.OCR2VRF:
+			filters, err = ocr2coordinator.FilterNamesFromSpec(spec)
+			if err != nil {
+				d.lggr.Errorw("failed to derive ocr2vrf filter names from spec", "err", err, "spec", spec)
+			}
+		case job.OCR2Keeper:
+			filters, err = ocr2keeper.FilterNamesFromSpec20(spec)
+			if err != nil {
+				d.lggr.Errorw("failed to derive ocr2keeper filter names from spec", "err", err, "spec", spec)
+			}
+		default:
+			return nil
 		}
-	default:
-		return nil
-	}
 
-	rargs := types.RelayArgs{
-		ExternalJobID: jb.ExternalJobID,
-		JobID:         spec.ID,
-		ContractID:    spec.ContractID,
-		New:           false,
-		RelayConfig:   spec.RelayConfig.Bytes(),
-	}
+		rargs := types.RelayArgs{
+			ExternalJobID: jb.ExternalJobID,
+			JobID:         spec.ID,
+			ContractID:    spec.ContractID,
+			New:           false,
+			RelayConfig:   spec.RelayConfig.Bytes(),
+		}
 
-	relayFilters, err := evmrelay.FilterNamesFromRelayArgs(rargs)
-	if err != nil {
-		d.lggr.Errorw("Failed to derive evm relay filter names from relay args", "err", err, "rargs", rargs)
-		return nil
-	}
+		relayFilters, err := evmrelay.FilterNamesFromRelayArgs(rargs)
+		if err != nil {
+			d.lggr.Errorw("Failed to derive evm relay filter names from relay args", "err", err, "rargs", rargs)
+			return nil
+		}
 
-	filters = append(filters, relayFilters...)
+		filters = append(filters, relayFilters...)
 
-	for _, filter := range filters {
-		d.lggr.Debugf("Unregistering %s filter", filter)
-		err = lp.UnregisterFilter(filter, pg.WithQueryer(q))
-		if err != nil {
-			return errors.Wrapf(err, "Failed to unregister filter %s", filter)
+		for _, filter := range filters {
+			d.lggr.Debugf("Unregistering %s filter", filter)
+			err = lp.UnregisterFilter(filter, pg.WithQueryer(q))
+			if err != nil {
+				return errors.Wrapf(err, "Failed to unregister filter %s", filter)
+			}
 		}
-	}
+	} // EVM specific cleanup
```

Comment:
Would it make sense to group this evm-only block in a helper method? Maybe not if too many in-scope variables would have to be passed along :balance_scale:

Reply:
yeah, thought about that too. done now